plugins/connectors/mongodb/config_test.go

```diff
@@ -0,0 +1,438 @@
+		t.Error("expected Pagination to be true")
+	}
+
+	if config.PageSize != 100 {
+		t.Errorf("expected PageSize to be 100, got %d", config.PageSize)
+	}
+}
+
+func TestConfigWithLastModifiedField(t *testing.T) {
+	config := &Config{
+		ConnectionURI:     "mongodb://localhost:27017/test",
+		Database:          "test",
+		LastModifiedField: "updated_at",
+		Collections: []CollectionConfig{
+			{
+				Name: "users",
+			},
+		},
+	}
+
+	plugin := &Plugin{}
+	err := plugin.validateConfig(config)
+	if err != nil {
+		t.Errorf("validateConfig() error = %v", err)
+	}
+
+	if config.LastModifiedField != "updated_at" {
+		t.Errorf("expected LastModifiedField to be 'updated_at', got %s", config.LastModifiedField)
+	}
+}
+
+func TestConfigWithAuthDatabase(t *testing.T) {
+	config := &Config{
+		ConnectionURI: "mongodb://user:pass@localhost:27017/test",
+		Database:      "test",
+		AuthDatabase:  "admin",
+		Collections: []CollectionConfig{
+			{
+				Name: "users",
+			},
+		},
+	}
+
+	plugin := &Plugin{}
+	err := plugin.validateConfig(config)
+	if err != nil {
+		t.Errorf("validateConfig() error = %v", err)
+	}
+
+	if config.AuthDatabase != "admin" {
+		t.Errorf("expected AuthDatabase to be 'admin', got %s", config.AuthDatabase)
+	}
+}
+
+func TestConfigWithClusterType(t *testing.T) {
+	tests := []struct {
+		name        string
+		clusterType string
+		wantErr     bool
+	}{
+		{"standalone", "standalone", false},
+		{"replica_set", "replica_set", false},
+		{"sharded", "sharded", false},
+		{"invalid", "invalid", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := &Config{
+				ConnectionURI: "mongodb://localhost:27017/test",
+				Database:      "test",
+				ClusterType:   tt.clusterType,
+				Collections: []CollectionConfig{
+					{
+						Name: "users",
+					},
+				},
+			}
+
+			plugin := &Plugin{}
+			err := plugin.validateConfig(config)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("validateConfig() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestAdvancedConfigOptions(t *testing.T) {
+	config := &Config{
+		ConnectionURI: "mongodb://localhost:27017/test",
+		Database:      "test",
+		Collections: []CollectionConfig{
+			{
+				Name: "users",
+			},
+		},
+		EnableProjection: false,
+		EnableIndexHint:  false,
+	}
+
+	plugin := &Plugin{}
+	plugin.setDefaultConfig(config)
+
+	// Test that advanced options are enabled by default
+	if !config.EnableProjection {
+		t.Error("expected EnableProjection to be enabled by default")
+	}
+
+	if !config.EnableIndexHint {
+		t.Error("expected EnableIndexHint to be enabled by default")
+	}
+
+	// Test with explicit values
+	config.EnableProjection = false
+	config.EnableIndexHint = false
+	plugin.setDefaultConfig(config)
+
+	if config.EnableProjection {
+		t.Error("expected EnableProjection to respect explicit false value")
+	}
+
+	if config.EnableIndexHint {
+		t.Error("expected EnableIndexHint to respect explicit false value")
+	}
+}
```

Comment:
I don't see value of these tests, if this is a integration test, you should verify the final docs synced from MongoDB, otherwise they are a waste of CI pipeline.